trans/train.py

```diff
@@ -322,15 +323,31 @@ def train_worker_init_fn(worker_id):
     best_epoch = 0
     patience = 0
 
+    sample_ids = list(range(len(training_data.samples)))
+    sample_stack = []
+
     for epoch in range(args.epochs):
 
         logging.info("Training...")
         transducer_.train()
         transducer_.zero_grad()
         with utils.Timer():
             train_loss = 0.
-            # rollin not implemented at the moment
-            # rollin = rollin_schedule(epoch)
+
+            #rollin
+            if args.rollin is not None and args.rollin:
+                rollin = rollin_schedule(epoch)
+                nr_samples = int(len(training_data.samples) * rollin)
+                rollin_samples = random.sample(sample_ids, nr_samples)
+                with torch.no_grad():
+                    # restore
+                    for id_, sample in sample_stack:
```

Comment:
@peter-makarov Here my idea was to sample an increasing size of samples each epoch while sampling a different subset each epoch (and setting the previous subset to the previous status (teacher forcing))

Reply:
Makes sense.